File: leetcode/985. Sum of Even Numbers After Queries/985.java
// https://helloacm.com/compute-the-sum-of-even-numbers-after-queries/
// https://leetcode.com/problems/sum-of-even-numbers-after-queries/

class Solution {
    public int[] sumEvenAfterQueries(int[] A, int[][] queries) {
        int num = queries.length;
        int[] sum = new int[num];
        int cursum = 0;
        for (int a: A) {
            if (a % 2 == 0) {
                cursum += a;
            }
        }
        for (int i = 0; i < num; ++ i) {
            int idx = queries[i][1];
            int val = queries[i][0];
            if (A[idx] % 2 == 0) cursum -= A[idx];
            A[idx] += val;
            if (A[idx] % 2 == 0) cursum += A[idx];
            sum[i] = cursum;
        }
        return sum;
    }
}

File: leetcode/985. Sum of Even Numbers After Queries/985.js
// https://helloacm.com/compute-the-sum-of-even-numbers-after-queries/
// https://leetcode.com/problems/sum-of-even-numbers-after-queries/

/**
 * @param {number[]} A
 * @param {number[][]} queries
 * @return {number[]}
 */
var sumEvenAfterQueries = function(A, queries) {
    var sum = [];
    var cursum = A.filter(x => x % 2 === 0).reduce( (a, b) => a + b, 0);
    for (var i = 0; i < queries.length; ++ i) {
        var idx = queries[i][1];
        var val = queries[i][0];
        if (A[idx] % 2 === 0) cursum -= A[idx];
        A[idx] += val;
        if (A[idx] % 2 === 0) cursum += A[idx];
        sum.push(cursum);
    }
    return sum;
};
